integreat_cms/firebase_api/firebase_data_client.py

```diff
@@ -0,0 +1,108 @@
+import logging
+from collections import defaultdict
+from datetime import datetime
+from typing import Any, Dict
+
+import requests
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+from integreat_cms.firebase_api.firebase_security_service import FirebaseSecurityService
+
+logger = logging.getLogger(__name__)
+
+
+# pylint: disable=too-few-public-methods
+class FirebaseDataCache:
+    """
+    Cache for the Firebase Cloud Messaging Data API Client
+    """
+
+    def __init__(self) -> None:
+        """
+        Initialize empty cache
+        """
+        self.data: Dict[Any, float | int] | None = None
+        self.timestamp: datetime | None = None
+
+    def is_valid(self) -> bool:
+        """
+        :return: Whether the cache contains a valid entry from today
+        """
+        if self.timestamp is None:
+            return False
+
+        return datetime.now().date() == self.timestamp.date()
```

Comment:
Is there an established method for handling caching in our system? So far, I've only found @cached_property, which differs slightly as it does not expire after a certain time.

Reply:
Well, I guess you could just use Django's inbuilt caching framework?

https://docs.djangoproject.com/en/5.0/topics/cache/#basic-usage

No need to re-implement it yourself.